librosa/sequence.py

```diff
@@ -359,6 +361,341 @@ def __dtw_backtracking(D_steps, step_sizes_sigma):  # pragma: no cover
+        New Journal of Physics 11, no. 9 (2009): 093017.
+
+    .. [2] Eckmann, J. P., S. Oliffson Kamphorst, and D. Ruelle.
+        "Recurrence plots of dynamical systems."
+        World Scientific Series on Nonlinear Science Series A 16 (1995): 441-446.
+
+
+    Parameters
+    ----------
+    sim : np.ndarray [shape=(N, M), non-negative]
+        The similarity matrix to use as input.
+
+        This can either be a recurrence matrix (self-similarity)
+        or a cross-similarity matrix between two sequences.
+
+    gap_onset : float > 0
+        Penalty for introducing a gap to an alignment sequence
+
+    gap_extend : float > 0
+        Penalty for extending a gap in an alignment sequence
+
+    knight_moves : bool
+        If `True` (default), allow for "knight moves" in the alignment,
+        e.g., `(n, m) => (n + 1, m + 2)` or `(n + 2, m + 1)`.
+
+        If `False`, only allow for diagonal moves `(n, m) => (n + 1, m + 1)`.
+
+    backtrack : bool
+        If `True`, return the alignment path.
+
+        If `False`, only return the score matrix.
+
+    Returns
+    -------
+    score : np.ndarray [shape=(N, M)]
+        The alignment score matrix.  `score[n, m]` is the cumulative value of
+        the best alignment sequence ending in frames `n` and `m`.
+
+    path : np.ndarray [shape=(k, 2)] (optional)
+        If `backtrack=True`, `path` contains a list of pairs of aligned frames
+        in the best alignment sequence.
+
+        `path[i] = [n, m]` indicates that row `n` aligns to column `m`.
+
+    See Also
+    --------
+    segment.recurrence_matrix
+    segment.cross_similarity
+    dtw
+
+    Examples
+    --------
+    Simple diagonal path enhancement (L-mode)
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> y, sr = librosa.load(librosa.util.example_audio_file(),
+    ...                      offset=10, duration=30)
+    >>> chroma = librosa.feature.chroma_cqt(y=y, sr=sr)
+    >>> # Use time-delay embedding to reduce noise
+    >>> chroma_stack = librosa.feature.stack_memory(chroma, n_steps=3)
+    >>> # Build recurrence, suppress self-loops within 1 second
+    >>> rec = librosa.segment.recurrence_matrix(chroma_stack, width=43,
+    ...                                         mode='affinity',
+    ...                                         metric='cosine')
+    >>> # using infinite cost for gaps enforces strict path continuation
+    >>> L_score, L_path = librosa.sequence.rqa(rec, np.inf, np.inf,
+    ...                                        knight_moves=False)
+    >>> plt.subplot(1,2,1)
+    >>> librosa.display.specshow(rec, x_axis='frames', y_axis='frames')
+    >>> plt.title('Recurrence matrix')
+    >>> plt.subplot(1,2,2)
+    >>> librosa.display.specshow(L_score, x_axis='frames', y_axis='frames')
+    >>> plt.plot(L_path[:, 1], L_path[:, 0], label='Optimal path', color='c')
+    >>> plt.legend()
+    >>> plt.show()
```

Comment:
can you make sure that the x-axis and y-axis have equal length? ("square" mode for specshow)

Reply:
specshow doesn't directly support aspect ratios, but i can change the figure size.